lib/streamlit/elements/media.py

```diff
@@ -138,26 +141,35 @@ def video(
             Includes support for YouTube URLs.
             Numpy arrays and raw data formats must include all necessary file
             headers to match specified file format.
+
         format : str
-            The mime type for the video file. Defaults to 'video/mp4'.
+            The mime type for the video file. Defaults to ``"video/mp4"``.
             See https://tools.ietf.org/html/rfc4281 for more info.
+
         start_time: int
             The time from which this element should start playing.
+
         subtitles: str, dict, or io.BytesIO
```

Comment:
nit: A `Path` object and just `bytes` are also supported here.

Reply:
For `st.audio`, we also include `file` as a type. Should that also be added to `st.video`? I think there are a few other places where we should add `Path`; I think we miss calling that out explicitly in favor of `str` to convey the path. From a quick search:
* `AppTest.from_file`
* `st.components.v1.declare_component`
* `st.switch_page`
* `st.page_link`
* `st.image`
* `st.chat_message`

Reply:
Regarding Path: I think the subtitle was the first instance where we started supporting this object. As follow-up, we should probably update all the other elements that use file paths to also support `Path` objects 👍 

Not sure about the `file` type, might be good to support that as well as follow up (@snehankekre).

Reply:
I've added `bytes` and `Path` to `subtitles`. I can follow up after release for further unification around `Path` and `file`.

Reply:
> As follow-up, we should probably update all the other elements 

100% in agreement. Let's do that as a project for the next two releases 😄 

>  I can follow up after release for further unification around Path and file.

@sfc-gh-dmatthews that would be great! But you'd have to wait for the st. commands to actually support `Path` objects before making the docstring edits.